lib/Doctrine/Migrations/Version.php

```diff
@@ -109,19 +104,46 @@ public function getConfiguration() : Configuration
         return $this->configuration;
     }
 
+    public function getMigration() : AbstractMigration
+    {
+        return $this->migration;
+    }
+
     public function isMigrated() : bool
     {
         return $this->configuration->hasVersionMigrated($this);
     }
 
-    public function markMigrated() : void
+    public function setState(int $state) : void
     {
-        $this->markVersion(self::DIRECTION_UP);
+        $this->state = $state;
     }
 
-    public function markNotMigrated() : void
+    public function getExecutionState() : string
     {
-        $this->markVersion(self::DIRECTION_DOWN);
+        switch ($this->state) {
+            case self::STATE_PRE:
+                return 'Pre-Checks';
+
+            case self::STATE_POST:
+                return 'Post-Checks';
+
+            case self::STATE_EXEC:
+                return 'Execution';
+
+            default:
+                return 'No State';
+        }
+    }
+
+    public function getTime() : ?float
+    {
+        return $this->time;
+    }
+
+    public function setTime(float $time) : void
```

Comment:
Instead of doing this, could it be moved to the Result? Or possibly passed around through some ExecutionContext?

Reply:
Yes, I think we can refactor this further to get rid of all the state in the Version class.